### PR TITLE
Sign up feature

### PR DESCRIPTION
**Pull request by pandwoter**
seems that work on that feature is done

### CODE REVIEW

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  def facebook
+    shared_login_logic "facebook"
+  end
+
+  def github
+    shared_login_logic "github"
+  end
+
+  def shared_login_logic service
```
Comment (dmokreckiy) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Use brackets in definitions, Luke

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    else
+      session["devise.user_attributes"] = request.env["omniauth.auth"]
+      redirect_to new_user_registration_url
+    end
+  end
+
+  def failure
+    redirect_to root_path
+  end
+end
```
Comment (dmokreckiy) [inline app/controllers/users/omniauth_callbacks_controller.rb]: New line missing

File: app/views/application/_appropriate_background_image.html.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<% if @user %>
+    <%= render 'motorist_background' %>
```
Comment (dmokreckiy) [inline app/views/application/_appropriate_background_image.html.erb]: Two spaces, right?

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 2 above)
Comment (dmokreckiy) [reply app/controllers/users/omniauth_callbacks_controller.rb]: Everywhere =)

File: app/views/devise/passwords/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+        <%= render "devise/shared/password_new_form" %>
+
+    </div>
+</div>
+
+
+
+
+
```
Comment (dmokreckiy) [inline app/views/devise/passwords/new.html.erb]: Too many empty lines.

File: config/environment.rb [hunk 5]
```diff
@@ -3,3 +3,13 @@
 
 # Initialize the Rails application.
 Rails.application.initialize!
+
+ActionMailer::Base.smtp_settings = {
+    :user_name => 'car_asistant_mailer',
```
Comment (dmokreckiy) [inline config/environment.rb]: You can use ENV vars for these credentials.
https://mkdev.me/posts/peremennye-okruzheniya-v-rails-dotenv-figaro-i-prochie
OR
https://www.engineyard.com/blog/rails-encrypted-credentials-on-rails-5.2

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 1 above)
Comment (pandwoter) [reply app/controllers/users/omniauth_callbacks_controller.rb]: fixed

File: app/controllers/users/omniauth_callbacks_controller.rb (hunk 2 above)
Comment (pandwoter) [reply app/controllers/users/omniauth_callbacks_controller.rb]: fixed

File: app/views/application/_appropriate_background_image.html.erb (hunk 3 above)
Comment (pandwoter) [reply app/views/application/_appropriate_background_image.html.erb]: fixed

File: app/views/devise/passwords/new.html.erb (hunk 4 above)
Comment (pandwoter) [reply app/views/devise/passwords/new.html.erb]: fixed

**Pull request merged**